### ISSUE TITLE
Gigablast engine breaks after a couple of hours

### ISSUE THREAD

**Issue by Elendil211** [searx/searx#3078]
**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**

```
git log -1
commit db2e8fd8b25f6f4bf9e1500589c9e48b8039b0cf (HEAD -> master, origin/master, origin/HEAD)
Author: searx-bot <69454556+searx-bot@users.noreply.github.com>
Date:   Mon Nov 15 20:32:34 2021 +0100

    Update searx.data - update_wikidata_units.py (#3050)

    Co-authored-by: dalf <dalf@users.noreply.github.com>

``` 

**How did you install Searx?**
I installed it following the wiki. I pulled the repository from git. Using apache server.

**What happened?**
Gigablast engine stops working after a couple hours. It produces the following error: `exception : Expecting value: line 1 column 1 (char 0)`

I observed the following behaviour:

1. I run
    ```
    sudo -H systemctl restart apache2
    sudo -H service uwsgi restart searx
    ```
2. First search fails with the error
3. Next few searches run well
4. After some time the error appears with every search. Repeat with step 1.

**How To Reproduce**
Install searx, enable gigablast, use it for a couple of hours.

**Expected behavior**
Gigablast engine should not create errors.

**Screenshots & Logs**

`/var/log/uwsgi/app/searx.log` is full with this error:

```
ERROR:searx.search.processor.online:engine gigablast : exception : Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "/usr/local/searx/searx-src/searx/search/processors/online.py", line 139, in search
    search_results = self._search_basic(query, params)
  File "/usr/local/searx/searx-src/searx/search/processors/online.py", line 123, in _search_basic
    return self.engine.response(response)
  File "/usr/local/searx/searx-src/searx/engines/gigablast.py", line 93, in response
    response_json = loads(resp.text)
  File "/usr/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Additional context**
 I'm using Raspbian on Raspberry Pi 4B
This seems to be the same issue as this one: https://github.com/searx/searx/issues/2141
I opened a new issue as the old one doesn't seem to get any attention.

**Comment by unixfox** [comment]
Did you test if you are blocked from gigablast when the error occurs?

Like go to https://gigablast.com and do a search, then check if everything is fine or not.

What keyword did you search when the error occurred?

**Comment by Elendil211** [comment]
I'm not blocked by gigablast. If I go directly to gigablast.com, the results are fine.

I can use literally any keyword. I just tried "test", "wiki", "gigablast", "youtube", "searx", and "searx gigablast crash". Every request crashes gigablast.

I'm under the impression (not entirely sure, because I don't use my instance alone), that the crash is not depending on how often gigablast was used for searches. I had it disabled by default, and it crashed on my first search after I enabled it locally. 

After I restarted with 
```
sudo -H systemctl restart apache2
sudo -H service uwsgi restart searx
```
everything was fine again. I used the exact same keywords, and I got results from gigablast.

Also to be noted: I just noticed, that the first search only fails, if the result page is the first page that is displayed after the restart. 
For example:

- I have the page loaded, and gigablast enabled locally (default disabled), I restart the server, make a search request -> crash
- I restart the server with gigablast enabled by default, type the search request in the URL bar -> crash
- I restart the server, go to the searx startpage, type the request in the field -> no crash
- I have the page loaded, restart the server, go to settings, make a search request -> no crash

An example `resp.text` when gigablast crashes looks like this:

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Gigablast Search Results</title>
<style><!--body {font-family:Arial, Helvetica, sans-serif;margin: 0px 0px;color: #000000;letter-spacing: 0.06em;
font-size: 14px;}
a {text-decoration:none;}
a:link {color:#00c}a:visited {color:#551a8b}a:active {color:#f00}.bold {font-weight: bold;}.bluetable {background:#d1e1ff;margin-bottom:15px;font-size:14px;}
.url {color:#008000;}
.cached, .cached a {font-size: 10px;color: #666666;}
table {font-family:Arial, Helvetica, sans-serif;color: #000000;font-size: 14px;}
table.result {font-family:Arial, Helvetica, sans-serif;line-height: 18px;color: #000000;font-size: 14px;background-color: white;border:2px solid #aaaaaa;}
table.kidresult {font-family:Arial, Helvetica, sans-serif;line-height: 18px;color: #000000;font-size: 14px;background-color: white;}
.title {font-family:Arial, Helvetica, sans-serif;line-height: 20px;font-size: 16px;}
.kidtitle {font-family:Arial, Helvetica, sans-serif;line-height: 18px;font-size: 14px;}
.mainurl {font-family:Arial, Helvetica, sans-serif;line-height: 18px;color: green;font-size: 14px;}
.kidurl {font-family:Arial, Helvetica, sans-serif;line-height: 16px;color: green;font-size: 12px;}
.pubdate {font-family:Arial, Helvetica, sans-serif;line-height: 18px;color: gray;font-size: 14px;}
.kidpubdate {font-family:Arial, Helvetica, sans-serif;line-height: 16px;color: gray;font-size: 12px;}
.bluetable {background:#d1e1ff;margin-bottom:15px;font-size:14px;}
.nav {font-size:20px;}
.directory {font-size: 14px;}
-->
</style>
</head>
<script>
<!--
var openmenu=''; var inmenuclick=0;var lasth=null;function x(){document.f.q.focus();}
// --></script>
<script>function show(id){var e = document.getElementById(id);if ( e.style.display == 'none' ){e.style.display = '';}else {e.style.display = 'none';}}</script><body id=body12 style=background-color:#e8e8e8;padding-left:10px;padding-right:10px; onLoad="var client = new XMLHttpRequest();
client.onreadystatechange = handler;
var uxrl='/search?c=main&format=json&q=wiki&dr=1&showgoodimages=0&qlangcountry=de&qlang=de&fromjs=1&rand=1638566456038&p';
uxrl=uxrl+'vvj=303768303';
client.open('GET', uxrl );
client.send();
"><script type="text/javascript">
function handler() {
if(this.readyState == 4 ) {
document.getElementById('bodycont').innerHTML=this.responseText;
}}
function ccc ( gn ) {
var e = document.getElementById('fd'+gn);
var f = document.getElementById('sd'+gn);
if ( e.style.display == 'none' ){
e.style.display = '';
f.style.display = 'none';
}
else {
e.style.display = 'none';
f.style.display = '';
}
}
</script>
<div id=bodycont><div style=padding-left:50px;padding-top:50px;><img height=50 width=50 src=/gears.gif><br/><br/><b>Waiting for results... </b><br/><br/></div><div id=placeholder style=min-height:100000px;></div></div>
<br/></body>
</html>
```

**Comment by kvch** [comment]
The error you see is due to the "speed" or the lack of speed in Gigablast. I added a new option to the engine, so a faster response can be configured to avoid these errors you are seeing.
